SMPyBandits/Policies/SWA.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+""" The Sliding Window Average policy for rotting bandits.
+Reference: [Levine , Crammer & Mannor, 2017].
+"""
+from __future__ import division, print_function  # Python 2 compatibility
+
+__author__ = "Julien Seznec"
+__version__ = "0.1"
+
+from math import sqrt, log
+import numpy as np
+import pandas as pd
+np.seterr(divide='ignore')  # XXX dangerous in general, controlled here!
+
+try:
+    from .IndexPolicy import IndexPolicy
+except ImportError:
+    from IndexPolicy import IndexPolicy
+
+
+class SWA(IndexPolicy):
+    """ The Sliding Window Average policy for rotting bandits.
+    Reference: [Levine , Crammer & Mannor, 2017].
+    """
+    def __init__(self, nbArms, horizon=1, subgaussian=1, maxDecrement=1,alpha=0.2, doublingTrick=False):
+        super(SWA, self).__init__(nbArms)
+        self.t = 0
+        self.nbArms = nbArms
+        self.arms_history = {}
+        for arm in range(nbArms):
+            self.arms_history[arm] = np.array([])
+        self.armSet = set(range(nbArms))
+        self.horizon = horizon
+        self.alpha = alpha if alpha else (2*maxDecrement)**(-2/3)
+        self.subgaussian = subgaussian
+        self.h = self.setWindow()
+        self.doubling = doublingTrick
+
+    def setWindow(self):
+        return self.alpha * (16 * self.subgaussian**2 * self.horizon**2 * log(sqrt(2)*self.horizon)/ len(self.armSet)**2)**(1/3)
+
+    def getReward(self, arm, reward):
+        super(SWA, self).getReward(arm, reward)
+        self.arms_history[arm] = np.insert(self.arms_history[arm], 0, 0) + reward
```

Comment:
Ok @JulSez that's the magic line right?
I wasn't sure what this line was doing. It makes more sense now.

Reply:
I don't know if it is magic but yes this is where every computation is done.